src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -520,7 +519,7 @@ abstract class RefChecks extends InfoTransform with scala.reflect.internal.trans
         }
         def checkOverrideTerm() {
           other.cookJavaRawInfo() // #2454
-          if (!overridesTypeInPrefix(lowType, highType, rootType)) { // 8
+          if (!overridesTypeInPrefix(lowType, highType, rootType, low.isModuleOrModuleClass || high.isModuleOrModuleClass)) { // 8
```

Comment:
Could you include a comment with the reasoning of grouping `low` & `high` together as far as them being a module? It would be good to extend test coverage to all possible combos (module overridden by val of module type, val of module type by module, module overridden by val of non-module type, val of non-module type by module).